src/generate-nestjs.ts

```diff
@@ -0,0 +1,205 @@
+import {
+  detectBatchMethod,
+  isEmptyType,
+  requestType,
+  responseObservable,
+  responsePromise,
+  responseType,
+  TypeMap,
+} from './types';
+import SourceInfo, { Fields } from './sourceInfo';
```

Comment:
This is just extracted functionality that used to live in the too-large `main.ts` file.